QuickOSM/core/process.py

```diff
@@ -115,31 +121,26 @@ def open_file(
 
             if output_dir:
                 dialog.set_progress_text(
-                    tr('From memory layer to GeoJSON: ' + layer))
+                    tr('From memory layer to file: ' + layer))
                 # Transforming the vector file
-                osm_geometries = {
-                    'points': QgsWkbTypes.Point,
-                    'lines': QgsWkbTypes.LineString,
-                    'multilinestrings': QgsWkbTypes.MultiLineString,
-                    'multipolygons': QgsWkbTypes.MultiPolygon
-                }
                 memory_layer = item['vector_layer']
 
-                encoding = get_default_encoding()
-                writer = QgsVectorFileWriter(
-                    outputs[layer],
-                    encoding,
-                    memory_layer.fields(),
-                    osm_geometries[layer],
-                    memory_layer.crs(),
-                    "GeoJSON")
-
-                for f in memory_layer.getFeatures():
-                    writer.addFeature(f)
-
-                del writer
+                options = QgsVectorFileWriter.SaveVectorOptions()
+                if output_format == Format.GeoPackage or output_format == Format.Kml:
+                    final_layer_name += '_' + layer
+                options.layerName = final_layer_name
+                options.driverName = output_format.value.driver_name
+                if os.path.exists(outputs[layer]):
+                    options.actionOnExistingFile = QgsVectorFileWriter.CreateOrOverwriteLayer
+                else:
+                    options.actionOnExistingFile = QgsVectorFileWriter.CreateOrOverwriteFile
+                context = QgsProject.instance().transformContext()
+                QgsVectorFileWriter.writeAsVectorFormatV2(
```

Comment:
Use `writeAsVectorFormat` to stay compatible with QGIS < 3.10

* v2 is only available in QGIS 3.10.3
* V3 is only available on QGIS 3.20

https://github.com/3liz/qgis-wfsOutputExtension/blob/master/wfsOutputExtension/wfs_filter.py#L172..L188